src/libraries/System.ComponentModel.Composition/tests/System/ComponentModel/Composition/Hosting/AssemblyCatalogTests.cs

```diff
@@ -36,10 +36,12 @@ public class TestAssemblyFour { }
 
     public class AssemblyCatalogTestsHelper
     {
+#pragma warning disable 618
```

Comment:
Once you apply `SYSLIB0012` as your `DiagnosticId`, you can change all of these to:
```suggestion
#pragma warning disable SYSLIB0012
```

Reply:
Updated to use `SYSLIB0012` as the pragma disable